File: scala/BeginnersSelection/02_PlacingMarbles.scala
// https://atcoder.jp/contests/abs/tasks/abc081_a

object Main {

  implicit class Pipeline[T](x: T) {
    def |>[S](f: T => S): S = f(x)
  }

  def main(args: Array[String]) = (
    io.StdIn.readLine
      .map(_.toString)
      .map(_.toInt)
      .sum
      |> println
  )
}

// https://atcoder.jp/contests/abs/submissions/19754090

File: kt/BeginnersSelection/02_PlacingMarbles.kt
// https://atcoder.jp/contests/abs/tasks/abc081_a

fun main(args: Array<String>) = (
  readLine()!!
    .map(Char::toString)
    .map(String::toInt)
    .sum()
    pp ::println
)

infix fun <T, R> T.pp(f: (T) -> R): R = f(this)

// https://atcoder.jp/contests/abs/submissions/20984377
